tools/link.py

```diff
@@ -2171,7 +2181,7 @@ def create_esm_wrapper(wrapper_file, support_target, wasm_target):
   mod = mod.replace('(import "wasi_snapshot_preview1"', f'(import "{support_url}"')
 
   wasm_as = os.path.join(building.get_binaryen_bin(), 'wasm-as')
-  shared.check_call([wasm_as, '--all-features', '-o', wasm_target, '-'], input=mod)
+  shared.check_call([wasm_as, '--all-features', '-g', '-o', wasm_target, '-'], input=mod)
```

Comment:
Do we always want `-g` here?

Reply:
Right now I'm doing this just to make a few more tests pass. In the long run we need to find a way to remove thie wasm-dis / wasm-as round-trip.

Reply:
I updated this so we only add `-g` when `EMIT_NAME_SECTION`